aip_urdf_compiler/scripts/compile_xacro.py

```diff
@@ -0,0 +1,484 @@
+    calib = Calibration(calib_yaml)
+
+    render_meta_data = {}
+    render_meta_data["default_config_path"] = f"$(find {project_name})/config"
+    render_meta_data["sensor_calibration_yaml_path"] = "$(arg config_dir)/sensors_calibration.yaml"
+    render_meta_data["sensor_units_includes"] = []
+    render_meta_data["sensor_units"] = []
+    render_meta_data["isolated_sensors_includes"] = []
+    render_meta_data["isolated_sensors"] = []
+
+    include_text = set()
+    sensor_items = []
+    for _, transform in calib.transforms.items():
+        link_type: LinkType = obtain_link_type(transform)
+        if link_type == LinkType.JOINT_UNITS:
+            render_meta_data["sensor_units_includes"].append(
+                link_dicts[link_type]["including_file"].format(filename=transform.name)
+            )
+            render_meta_data["sensor_units"].append(
+                {
+                    "base_frame": transform.base_frame,
+                    "child_frame": transform.child_frame,
+                    "macro_name": f"{transform.name}_macro",
+                    "name": transform.name,
+                }
+            )
+        else:
+            include_text.add(link_dicts[link_type]["including_file"])
+            sensor_items.append(link_dicts[link_type]["string_api"](transform))
+
+    render_meta_data["isolated_sensors_includes"] = list(include_text)
+    render_meta_data["isolated_sensors"] = sensor_items
+
+    rendered = base_template.render(render_meta_data)
+    print(rendered)
+
+    print("=====================================")
+    # Save the rendered template
+    with open(os.path.join(output_directory, "sensors.xacro"), "w") as file:
+        file.write(rendered)
+
+    # Write Sensor Units into separate files
+    with open(os.path.join(template_directory, "sensor_unit.xacro.template"), "r") as file:
+        sensor_units_template = Template(file.read())
+
+    for i, sensor_unit in enumerate(render_meta_data["sensor_units"]):
+        sensor_unit_calib_path = os.path.join(
+            calibration_directory, f"{sensor_unit['name']}_calibration.yaml"
+        )
+        sensor_unit_calib_yaml = load_yaml(sensor_unit_calib_path)
+        sensor_unit_calib = Calibration(sensor_unit_calib_yaml)
+        sensor_unit_render_meta_data = {}
+        sensor_unit_render_meta_data["unit_macro_name"] = sensor_unit["macro_name"]
+        sensor_unit_render_meta_data["default_config_path"] = render_meta_data[
+            "default_config_path"
+        ]
+        sensor_unit_render_meta_data["joint_unit_name"] = sensor_unit["name"]
+        sensor_unit_render_meta_data["current_base_link"] = sensor_unit_calib.base_frame
+        sensor_unit_isolated_sensors = []
+        for _, transform in sensor_unit_calib.transforms.items():
+            link_type: LinkType = obtain_link_type(transform)
+            include_text.add(link_dicts[link_type]["including_file"])
+            print(transform.name)
```

Comment:
Nits: I prefer to print the final result than to print the name iteratively.

Reply:
When we execute `colocn build` , we can see the print result in `pilot-auto.xx1/log/build_2024-12-23_hh-mm-ss/aip_xx1_description/streams.log`

For example, as follows. 

```
[0.102s]             ${calibration['base_link']['velodyne_rear_base_link']['pitch']}
[0.102s]             ${calibration['base_link']['velodyne_rear_base_link']['yaw']}"
[0.102s]     />
[0.102s]     </xacro:VLP-16>
[0.102s]   
[0.102s] </robot>
[0.102s] =====================================
[0.105s] camera0/camera
[0.105s] camera1/camera
[0.105s] camera2/camera
[0.105s] camera3/camera
[0.105s] camera4/camera
[0.105s] camera5/camera
[0.105s] camera6/camera
[0.105s] camera7/camera
[0.105s] velodyne_top
[0.106s] velodyne_left
[0.106s] velodyne_right
[0.106s] gnss
[0.106s] tamagawa/imu
[0.106s] <?xml version="1.0"?>
[0.106s] <robot xmlns:xacro="http://ros.org/wiki/xacro">
[0.106s]   <xacro:macro name="sensor_kit_macro" params="parent x y z roll pitch yaw">

```

Reply:
Thank you! [ddbc8f7](https://github.com/tier4/aip_launcher/pull/361/commits/ddbc8f7ec86070f18ba29829bc6ff740505e3c83) is easier to read.